File: client/src/app/models/deck-card.ts
export interface DeckCard {
  readonly id: number;
  readonly expression: string;
  readonly exampleSentence: string;
  readonly translation: string;
  readonly readingHiragana: string;
  readonly partOfSpeech: string;
  readonly furiganaSentence: string;
  readonly translatedSentence: string;
  readonly hasKanji: boolean;
}

export interface DeckPreview {
  readonly cards: readonly DeckCard[];
  readonly rejectedRows: readonly RejectedDeckRow[];
}

export interface RejectedDeckRow {
  readonly lineNumber: number;
  readonly offendingValue: string;
  readonly expectedShape: string;
}

File: server/TwoKSixKPractice.Api/Decks/DeckCard.cs
namespace TwoKSixKPractice.Api.Decks;

public sealed record DeckCard(
    int Id,
    string Expression,
    string ExampleSentence,
    string Translation,
    string ReadingHiragana,
    string PartOfSpeech,
    string FuriganaSentence,
    string TranslatedSentence,
    bool HasKanji);
